new-site/docs-site/content/0.19.0/guide/typesense-dynamodb.md

```diff
@@ -0,0 +1,240 @@
+
+* AmazonDynamoDBFullAccess
+* AmazonDynamoDBFullAccesswithDataPipeline
+* AWSLambdaBasicExecutionRole
+
+These IAM role permissions are just examples for the purposes of this guide. Before deploying for production, please consult the IAM documentation to only grant the minimal permissions needed for your particular use case.
+
+You can also do this using AWS CLI:
+
+Create a file named ```trust-relationship.json``` with the following contents.
+```json
+{
+   "Version": "2012-10-17",
+   "Statement": [
+     {
+       "Effect": "Allow",
+       "Principal": {
+         "Service": "lambda.amazonaws.com"
+       },
+       "Action": "sts:AssumeRole"
+     }
+   ]
+}
+```
+
+Then, create execute the following command
+
+```bash
+aws iam create-role --role-name TypesenseLambdaRole \
+    --path "/service-role/" \
+    --assume-role-policy-document file://trust-relationship.json
+```
+
+Now, create ```role-policy.json``` with the following contents. (Replace ```accountID``` and ```region```)
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "lambda:InvokeFunction",
+            "Resource": "arn:aws:lambda:region:accountID:function:typesense-indexing*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:region:accountID:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeStream",
+                "dynamodb:GetRecords",
+                "dynamodb:GetShardIterator",
+                "dynamodb:ListStreams"
+            ],
+            "Resource": "arn:aws:dynamodb:region:accountID:table/typesense/stream/*"
+        },
+    ]
+}
+```
+
+The policy has three statements that allow TypesenseLambdaRole to do the following:
+
+* Run a Lambda function ```typesense-indexing```. We'll be creating the function later in this tutorial.
+* Access Amazon CloudWatch Logs. The Lambda function writes diagnostics to CloudWatch Logs at runtime.
+* Read data from the DynamoDB stream for ```typesense```.
+
+Now, we are going to attach the above roles to our IAM execution role which we have created
+
+```bash
+aws iam put-role-policy --role-name TypesenseLambdaRole \
+    --policy-name TypesenseLambdaRolePolicy \
+    --policy-document file://role-policy.json
+```
+## Step 4: Create a Lambda Function
+
+Head over to Lambda section of AWS and create a new Lambda function with the above created execution role.
+
+Now, let's create a new python script file named `lambda_function.py`. For creating collections and indexing into your cloud cluster refer: [Typesense API](../api/README.md)
+
+```python
```

Comment:
Could you combine this snippet with the next one, so the entire code snippet that a user would write inside the lamdba is in one place?